File: SRXLite/Classes/UsageLog.vb
Option Strict On

Imports System.Data.SqlClient
Imports SRXLite.DataAccess
Imports SRXLite.Modules

Namespace Classes

    Public Class UsageLog

        Private _context As HttpContext

#Region " Constructors "

        ''' <summary>
        '''
        ''' </summary>
        ''' <param name="context"></param>
        ''' <remarks></remarks>
        Public Sub New(ByVal context As HttpContext)
            _context = context
        End Sub

#End Region

#Region " UsageData "

        Public Class UsageData
            Public ActionType As DataTypes.ActionType?
            Public BatchID As Integer?
            Public DocID As Long?
            Public DocPageID As Long?
            Public DocStatus As DataTypes.DocumentStatus?
            Public GroupID As Long?
            Public IPAddress As String
            Public SubuserID As Integer?
            Public UserID As Short
        End Class

#End Region

#Region " Insert "

        Public Sub Insert(ByVal data As UsageData)
            Dim command As New SqlCommand
            command.CommandText = "dsp_UsageLog_Insert"
            command.CommandType = CommandType.StoredProcedure
            With command.Parameters
                .Add(getSqlParameter("@UserID", data.UserID))
                .Add(getSqlParameter("@SubuserID", data.SubuserID))
                .Add(getSqlParameter("@ActionTypeID", data.ActionType))
                .Add(getSqlParameter("@DocStatusID", data.DocStatus))
                .Add(getSqlParameter("@DocID", data.DocID))
                .Add(getSqlParameter("@DocPageID", data.DocPageID))
                .Add(getSqlParameter("@GroupID", data.GroupID))
                .Add(getSqlParameter("@BatchID", data.BatchID))
                .Add(getSqlParameter("@ClientIP", GetClientIP(_context)))
            End With

            DB.ExecuteNonQuery(command)
        End Sub

#End Region

    End Class

End Namespace

File: SRXLite.CS/Classes/UsageLog.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Web;
using SRXLite.DataAccess;
using SRXLite.DataTypes;
using static SRXLite.Modules.Util;
using static SRXLite.DataAccess.DB;

namespace SRXLite.Classes
{
    public class UsageLog
    {
        private HttpContext _context;

        #region Constructors

        /// <summary>
        /// Constructor with HTTP context
        /// </summary>
        /// <param name="context"></param>
        public UsageLog(HttpContext context)
        {
            _context = context;
        }

        #endregion

        #region UsageData

        public class UsageData
        {
            public ActionType? ActionType { get; set; }
            public int? BatchID { get; set; }
            public long? DocID { get; set; }
            public long? DocPageID { get; set; }
            public DocumentStatus? DocStatus { get; set; }
            public long? GroupID { get; set; }
            public string IPAddress { get; set; }
            public int? SubuserID { get; set; }
            public short UserID { get; set; }
        }

        #endregion

        #region Insert

        public void Insert(UsageData data)
        {
            SqlCommand command = new SqlCommand();
            command.CommandText = "dsp_UsageLog_Insert";
            command.CommandType = CommandType.StoredProcedure;
            command.Parameters.Add(GetSqlParameter("@UserID", data.UserID));
            command.Parameters.Add(GetSqlParameter("@SubuserID", data.SubuserID));
            command.Parameters.Add(GetSqlParameter("@ActionTypeID", (int?)data.ActionType));
            command.Parameters.Add(GetSqlParameter("@DocStatusID", (byte?)data.DocStatus));
            command.Parameters.Add(GetSqlParameter("@DocID", data.DocID));
            command.Parameters.Add(GetSqlParameter("@DocPageID", data.DocPageID));
            command.Parameters.Add(GetSqlParameter("@GroupID", data.GroupID));
            command.Parameters.Add(GetSqlParameter("@BatchID", data.BatchID));
            command.Parameters.Add(GetSqlParameter("@ClientIP", GetClientIP(_context)));

            DB.ExecuteNonQuery(command);
        }

        #endregion
    }
}
